src/main/java/org/openbase/bco/bcozy/view/generic/WidgetPane.java

```diff
@@ -143,12 +144,25 @@ public void changed(ObservableValue<? extends Boolean> observable, Boolean lastA
                         currentTask.cancel(true);
                     }
                     currentTask = applyPrimaryActivationUpdate(newActivation);
-                } catch (CouldNotPerformException ex) {
+                    currentTask.get();
+                } catch (CouldNotPerformException | ExecutionException | InterruptedException ex) {
                     ExceptionPrinter.printHistory("Could not apply activation update " + this, ex, LOGGER);
+
+                    // If the task failed, reset the activation value.
+                    setPrimaryActivationWithoutNotification(lastActivation);
                 }
             }
         };
         primaryActivationProperty().addListener(primaryActivationObserver);
+
+        // @cromankiewicz: Without this primitive dummy listener, the toggleSwitch does not seem to
+        // be notified if the first try of the primary action fails. Don't ask me why.
+        primaryActivationProperty().addListener(new ChangeListener<Boolean>() {
+            @Override
+            public void changed(ObservableValue<? extends Boolean> observable, Boolean oldValue, Boolean newValue) {
+                System.out.println("old: " + oldValue + ", new: " + newValue);
```

Comment:
but even than, avoid spam on system out ;) At least use a logger and switch to the debug channel.